### PR TITLE
Add additional methods to BitcoinHelper

### PR DESCRIPTION
**Pull request by lukasz-zimnoch**
Refs: https://github.com/keep-network/tbtc/issues/703

Implemented some additional methods exposed by the Electrum Client which will be needed by the end-to-end test script for testnet: https://github.com/keep-network/local-setup/pull/20

### CODE REVIEW

File: src/BitcoinHelpers.js [hunk 1]
```diff
@@ -523,6 +523,55 @@ const BitcoinHelpers = {
...
+        const script = BitcoinHelpers.Address.toScript(bitcoinAddress)
+        return BitcoinHelpers.Transaction.findAllUnspentWithClient(electrumClient, script)
+      })
+    },
+    /**
+     * Gets the confirmed balance of the `bitcoinAddress`.
+     *
+     * @param {string} bitcoinAddress Bitcoin address to check.
+     *
+     * @return {Promise<Number>} A promise to the confirmed balance as satoshis.
```
Comment (Shadowfiend) [inline src/BitcoinHelpers.js:545]: in satoshis*

File: src/BitcoinHelpers.js [hunk 2]
```diff
@@ -523,6 +523,55 @@ const BitcoinHelpers = {
...
+     * @param {number} [includeWithinBlocks=1] - The estimate is based on having
+     *        the transaction included within this number of blocks. Defaults to
+     *        trying to include the transaction within 1 block.
+     *
+     * @return {Promise<number>} The estimated fee to execute the provided
+     *         transaction. Fails if the fee cannot be estimated.
+     */
+    estimateFeePerKb: async function(includeWithinBlocks) {
+      return BitcoinHelpers.withElectrumClient(async electrumClient => {
+        const feePerKb = await electrumClient.getFeeEstimate(includeWithinBlocks || 1)
```
Comment (Shadowfiend) [inline src/BitcoinHelpers.js:566]: Worth noting, this was always returning 0 in my deposit submitter experiments, which is why I ended up always using the minimum relay fee in [the other PR's implementation](https://github.com/keep-network/tbtc.js/pull/38/files#diff-d8016928a1a8bbd777a2a7d57dc8cbbeR181).

**Comment by lukasz-zimnoch** [comment]
Thanks for a turbo review @Shadowfiend!. Yep, it's still a draft because I'm still testing everything and there is still a possibility I will change something. Once I'm done with testing I'll make it ready for review and ping you.

File: src/BitcoinHelpers.js (hunk 2 above)
Comment (lukasz-zimnoch) [reply src/BitcoinHelpers.js:566]: Yep, I have the same. I thought there will be a difference if I use it against BTC testnet but nope. I will remove this method and use the default ones or the minimum relay fee in my test script.

**Comment by lukasz-zimnoch** [comment]
@Shadowfiend I'm ready here.

File: src/BitcoinHelpers.js [hunk 3]
```diff
@@ -542,36 +542,14 @@ const BitcoinHelpers = {
...
-      return BitcoinHelpers.withElectrumClient(async electrumClient => {
-        const feePerKb = await electrumClient.getFeeEstimate(includeWithinBlocks || 1)
-
-        if (feePerKb < 0) {
-          throw new Error(`Fee cannot be estimated; Electrum returned ${feePerKb}.`)
-        }
-
-        return feePerKb
-      })
-    },
```
Comment (Shadowfiend) [inline src/BitcoinHelpers.js]: I'm definitely interested in making this work in the future, btw. Need to see what about our node is making the server side of this fall over.

**Pull request merged**